client/src/features/editor/components/TextOptionModal/TextOptionModal.tsx

```diff
@@ -0,0 +1,209 @@
+import { Char } from "@noctaCrdt/Node";
+import { motion } from "framer-motion";
+import { useEffect, useRef, useState } from "react";
+import { createPortal } from "react-dom";
+import { modalContainer, optionButton, optionModal } from "./TextOptionModal.style";
+
+interface SelectionModalProps {
+  selectedNodes: Array<Char> | null;
+  isOpen: boolean;
+  onBoldSelect: () => void;
+  onItalicSelect: () => void;
+  onUnderlineSelect: () => void;
+  onStrikeSelect: () => void;
+  onClose: () => void;
+}
+
+const ModalPortal = ({ children }: { children: React.ReactNode }) => {
+  return createPortal(children, document.body);
+};
+
+export const TextOptionModal = ({
+  selectedNodes,
+  isOpen,
+  onBoldSelect,
+  onItalicSelect,
+  onUnderlineSelect,
+  onStrikeSelect,
+  onClose,
+}: SelectionModalProps) => {
+  const modalRef = useRef<HTMLDivElement>(null);
+  const [TextModalPosition, setTextModalPosition] = useState<{ top: number; left: number }>({
+    top: 0,
+    left: 0,
+  });
+  const [styleState, setStyleState] = useState({
+    isBold: false,
+    isItalic: false,
+    isUnderline: false,
+    isStrike: false,
+  });
+
+  useEffect(() => {
+    if (!isOpen) return;
+
+    const updateModalPosition = () => {
+      const selection = window.getSelection();
+      if (!selection || selection.isCollapsed) {
+        onClose();
+        return;
+      }
+
+      const range = selection.getRangeAt(0);
+      const rect = range.getBoundingClientRect();
+
+      setTextModalPosition({
+        top: rect.top - 40,
+        left: rect.left + rect.width / 2,
+      });
+    };
+
+    const handleClickOutside = (e: MouseEvent) => {
+      if (modalRef.current && !modalRef.current.contains(e.target as Node)) {
+        onClose();
+      }
+    };
+
+    updateModalPosition();
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+      setStyleState({
+        isBold: false,
+        isItalic: false,
+        isUnderline: false,
+        isStrike: false,
+      });
+    };
+  }, [isOpen, onClose]);
+
+  useEffect(() => {
+    if (!selectedNodes || selectedNodes.length === 0) {
+      setStyleState({
+        isBold: false,
+        isItalic: false,
+        isUnderline: false,
+        isStrike: false,
+      });
+      return;
+    }
+
+    // 각 스타일의 출현 횟수를 계산
+    const styleCounts = {
+      bold: 0,
+      italic: 0,
+      underline: 0,
+      strike: 0,
+    };
+
+    // 각 노드의 스타일을 확인하고 카운트
+    selectedNodes.forEach((node) => {
+      // node.style이 undefined인 경우 빈 배열로 처리
+      const nodeStyles = Array.isArray(node.style) ? node.style : node.style ? [node.style] : [];
+
+      if (nodeStyles.includes("bold")) styleCounts.bold += 1;
+      if (nodeStyles.includes("italic")) styleCounts.italic += 1;
+      if (nodeStyles.includes("underline")) styleCounts.underline += 1;
+      if (nodeStyles.includes("strike")) styleCounts.strike += 1;
```

Comment:
```suggestion
  if (nodeStyles.includes("strikethrough")) styleCounts.strike += 1;
```
으로 변경해야할 것 같아요! strikethrough으로 이름을 지으셔서, 스트라이크 부분만 선택된 효과가 안나오더라구요!

Reply:
이부분 수정해서 push했습니다! 감사합니다!